targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/device/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+function(_mbed_get_assembly_nrf52840_dk)
+    if(${MBED_TOOLCHAIN} STREQUAL "GCC_ARM")
+        set(STARTUP_FILE TOOLCHAIN_GCC_ARM/startup_NRF52840.S)
+    elseif(${MBED_TOOLCHAIN} STREQUAL "ARM")
+        set(STARTUP_FILE TOOLCHAIN_ARM_STD/startup_nrf52840.S)
+    elseif(${MBED_TOOLCHAIN} STREQUAL "IAR")
+        set(STARTUP_FILE TOOLCHAIN_IAR/startup_NRF52840_IAR.S)
+    endif()
+    target_sources(mbed-core INTERFACE ${STARTUP_FILE})
+endfunction()
```

Comment:
I've missed this one (@hugueskamba you pinpointed it but my search failed anyway). I'll send PR now to fix this (not blocking this PR anyway).